lib/types/application/ApplicationFormatter.js

```diff
@@ -40,15 +50,46 @@ class ApplicationFormatter extends AbstractUi5Formatter {
 		return p.join(projectPath, this._project.resources.pathMappings["/"]);
 	}
 
+	/**
+	 * Determine application namespace either based on a project`s
+	 * manifest.json or manifest.appdescr_variant (fallback if present)
+	 *
+	 * @returns {string} Namespace of the project
+	 * @throws {Error} if namespace can not be determined
+	 */
+	async getNamespace() {
+		try {
+			return await this.getNamespaceFromManifestJson();
+		} catch (manifestJsonError) {
+			if (manifestJsonError.code !== "ENOENT") {
+				throw manifestJsonError;
+			}
+			// No manifest.json present
+			// => attempt fallback to manifest.appdescr_variant
+			try {
+				return await this.getNamespaceFromManifestAppDescVariant();
+			} catch (appDescVarError) {
+				if (appDescVarError.code === "ENOENT") {
+					// Fallback not possible: No manifest.appdescr_variant present
+					// => Throw error indicating missing manifest.json (do not mention manifest.appdescr_variant)
+					throw new Error(
+						`Could not find required manifest.json for project ` +
```

Comment:
Should it be extended also as "Could not find required manifest.json or manifest.appdescr_variant for project ..."?

Reply:
Since `manifest.json` is still the file we expect in the majority of application projects (i.e. all but App Variants), I would rather not mention `manifest.appdescr_variant` if both are missing. 

I don't want to confuse new UI5 developers that just get started with their application and often find themselves missing a `manifest.json`. Because, adding a `manifest.appdescr_variant` in that case would not be the right approach.